### PR TITLE
feat: expose asset details

### PR DESCRIPTION
**Pull request by adrianhopebailie**
- Add a new control frame to exchange code and scale of remote account asset
 - Add logic to exchange data on new connection
 - Expose local and remote asset details
 - Add tests

### CODE REVIEW

**Comment by emschwartz** [comment]
Why does this need to be on the stream level, as opposed to part of a protocol built on top?

**Comment by emschwartz** [comment]
(I think of stream as being about getting the money and data there while figuring out how fast it can get it there. Information about the receiver feels like it should be another protocol on top of this one)

**Comment by adrianhopebailie** [comment]
> Information about the receiver feels like it should be another protocol on top of this one

Agree to a point. Asset and scale are inseparably linked to amount and address. An amount sent to/from an address has no meaning unless that amount has an asset and scale.

Also, imagine your application code has a handle to an instance of a STREAm server. It handles new incoming connections and exchanges data with the remote STREAM endpoint using a new "higher layer" protocol to determine the asset code and scale of the remote endoint.

How does it share it's own asset code and scale with the remote endpoint? It has no way of figuring out what asset it is actually sending? Likewise for the remote entity. This data is only available when the endpoint first initialises the local plugin via ILDCP.

The STREAM layer should record this (just like it does the address) and make these available as read-only properties to the higher layers.

**Comment by adrianhopebailie** [comment]
Having reviewed again, the code-smell that suggests this is the right thing to do are these lines:

 - https://github.com/interledgerjs/ilp-protocol-stream/pull/69/files#diff-f41e9d04a45c83f3b6f6e630f10117feL30
 - https://github.com/interledgerjs/ilp-protocol-stream/pull/69/files#diff-f41e9d04a45c83f3b6f6e630f10117feL134

We're using ILDCP to get info about the underlying account and then throwing most of that info away.

**Comment by kincaidoneil** [comment]
> How does it share it's own asset code and scale with the remote endpoint? It has no way of figuring out what asset it is actually sending? Likewise for the remote entity. This data is only available when the endpoint first initialises the local plugin via ILDCP.

Would it work to run connect on the plugin outside of STREAM, fetch the local asset scale & code via ILDCP, and when the secret and address are exchanged over SPSP, also share the asset scale & code, and then instantiate the STREAM server/create a connection?

**Comment by kincaidoneil** [comment]
Adding on to what you pointed out--maybe the `generateAddressAndSecret` method should also return the asset scale and code from those ILDCP requests (so it doesn't necessitate multiple requests)?

File: test/connection.test.ts [hunk 1]
```diff
@@ -37,17 +37,39 @@ describe('Connection', function () {
     this.serverConn = await connectionPromise
     this.serverConn.on('stream', (stream: DataAndMoneyStream) => {
       stream.setReceiveMax(10000)
-    })
```
Comment (emschwartz) [inline test/connection.test.ts]: Nit: trailing space

**Pull request merged**